.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,27 @@
+# This file is automatically added by @npmcli/template-oss. Do not edit.
+
+name: Pull Request Linting
+
+on:
+  pull_request:
+    types: [opened, reopened, edited, synchronize]
+
+jobs:
+  check:
+    name: Check PR Title or Commits
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Docs says this fetches all history for tags and branches. Is that necessary for commitlint to work properly?

Reply:
Yep! This was a fun one to debug. Without this, all that is checked out is a completely detached merge commit and we don't have access to the actual commits themselves to lint the titles of with `--from` and `--to` down on line 27.

Reply:
I did some digging and there is no middle ground. We either pull everything or nothing. There's no way to just pull the commits in the PR.